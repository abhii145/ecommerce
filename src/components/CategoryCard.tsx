import React from "react"
import { Link } from "react-router"
import { slugToSvg } from "../utils/categoryImages"

interface CategoryCardProps {
  title: string
  slug: string
}

const CategoryCard: React.FC<CategoryCardProps> = ({ title, slug }) => {
  const SvgImage = slugToSvg[slug]

  return (
    <>
      <Link
        to={`/products?category=${encodeURIComponent(slug)}`}
        className="block"
      >
        <div className="relative overflow-hidden rounded-lg shadow-lg">
          <img
            src={SvgImage}
            alt={title}
            className="w-full h-48 object-contain"
            loading="lazy"
          />
          <div className="absolute inset-0 bg-black bg-opacity-50 flex items-center justify-center">
            <h2 className="text-white text-xl font-bold">{title}</h2>
          </div>
        </div>
      </Link>
    </>
  )
}

export default CategoryCard
