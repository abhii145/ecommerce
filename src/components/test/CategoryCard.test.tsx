import { render, screen } from "@testing-library/react"
import { BrowserRouter } from "react-router"
import CategoryCard from "../CategoryCard"

jest.mock("../../utils/categoryImages", () => ({
  slugToSvg: {
    "category-1": "path/to/category-1-image.svg",
    "category-2": "path/to/category-2-image.svg",
  },
}))

describe("CategoryCard", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <CategoryCard title="Category 1" slug="category-1" />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("renders without crashing", () => {
    render(
      <BrowserRouter>
        <CategoryCard title="Category 1" slug="category-1" />
      </BrowserRouter>
    )

    expect(screen.getByText("Category 1")).toBeInTheDocument()
  })

  it("displays the correct image based on the slug", () => {
    render(
      <BrowserRouter>
        <CategoryCard title="Category 1" slug="category-1" />
      </BrowserRouter>
    )

    const imgElement = screen.getByAltText("Category 1") as HTMLImageElement
    expect(imgElement.src).toContain("path/to/category-1-image.svg")
  })

  it("renders the title correctly", () => {
    render(
      <BrowserRouter>
        <CategoryCard title="Category 1" slug="category-1" />
      </BrowserRouter>
    )

    const titleElement = screen.getByText("Category 1")
    expect(titleElement).toBeInTheDocument()
    expect(titleElement).toHaveClass("text-white text-xl font-bold")
  })
})
