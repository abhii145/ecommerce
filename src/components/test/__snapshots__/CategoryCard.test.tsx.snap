// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="block"
    data-discover="true"
    href="/products?category=category-1"
  >
    <div
      class="relative overflow-hidden rounded-lg shadow-lg"
    >
      <img
        alt="Category 1"
        class="w-full h-48 object-contain"
        loading="lazy"
        src="path/to/category-1-image.svg"
      />
      <div
        class="absolute inset-0 bg-black bg-opacity-50 flex items-center justify-center"
      >
        <h2
          class="text-white text-xl font-bold"
        >
          Category 1
        </h2>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
